01/app.js

```diff
@@ -1,15 +1,26 @@
 function getProductList() {
-    return [
-        {name: 'JavaScript: podstawy', type: 'book', count: 3, price: 67.19, discount: 0.21},
-        {name: 'React: podstawy', type: 'book', count: 4, price: 79.17, discount: 0.27}
+    return [{
+            name: 'JavaScript: podstawy',
+            type: 'book',
+            count: 3,
+            price: 67.19,
+            discount: 0.21
+        },
+        {
+            name: 'React: podstawy',
+            type: 'book',
+            count: 4,
+            price: 79.17,
+            discount: 0.27
+        }
     ]
 }
 
 function getTotalPrice(productList) {
-    return productList.reduce( (product, acc) => {
-        return acc + (product.price * product.count);
-    });
+    return productList.reduce((acc, product) => {
+        return acc + ((product.price - product.discount) * product.count);
```

Comment:
`product.discount` to procentowy rabat więc tutaj obliczenie powinno być inne ;)

Reply:
Poprawione